### PR TITLE
Fix an ArrayOutOfBoundException with array:get

### PR DESCRIPTION
**Pull request by adamretter**
Closes https://github.com/eXist-db/exist/issues/2317

### CODE REVIEW

File: test/src/xquery/arrays/arrays.xql [hunk 1]
```diff
@@ -134,6 +134,14 @@ function arr:function-item-invalid() {
         $a("x")
 };
 
+declare
```
Comment (dizzzz) [inline test/src/xquery/arrays/arrays.xql]: what about testing [0] [-1] [1] and [3] ? these are cornercases...

Comment (adamretter) [reply test/src/xquery/arrays/arrays.xql]: I have now incorporated your further tests.

Comment (dizzzz) [reply test/src/xquery/arrays/arrays.xql]: but the logic is still not throwing the correct errors, right?

Comment (adamretter) [reply test/src/xquery/arrays/arrays.xql]: I added further tests. The logic was correct, but the error code was wrong, I have corrected that now.

**Comment by adamretter** [comment]
@dizzzz the error was thrown, it just had the wrong code! I have fixed this and rebased, hopefully good to go now.

**Pull request merged**